### PR TITLE
Development

### PR DESCRIPTION
**Pull request by 300ms**
Waiting for a code review

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+def bubble_sort(arr)
+  arr.each {
+    x = 0
```
Comment (stratospherique) [inline bubble_sort.rb]: - You can use each block in this manner 
```ruby
    arr.each |item| do
        # logic
    end
```

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    while x < arr.length - 1
+      if arr[x].length > arr[x + 1].length
+        tmp = arr[x + 1]
+        arr[x + 1] = arr[x]
+        arr[x] = tmp
+      end
+      x += 1
+    end
+  }
+end
```
Comment (stratospherique) [inline bubble_sort.rb:35]: * [x] bubble_sort_by raises an error if block is not given. 
- **[HINT]** Check for `block_given?` at the beginning of the method and raise an error if no block was passed.

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great Progress :clap: However, further changes needs to be applied in order to get approval for this project. check :arrow_down: 

- **[REQUIRED]** Check the code comments below, they should help you refactor your code :nerd_face: 

- **[REQUIRED]** Fix linter checks errors, I included a suggestion in the code comments. They should be helpful.

### That's it :relaxed: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: You did it :smile: 

You can merge it :+1: 
Good luck with the next project

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Pull request merged**